From ff63ad7fff9dee3df228455bbc69bd39d2572c86 Mon Sep 17 00:00:00 2001
From: jinghua <xigu@marvell.com>
Date: Tue, 29 Nov 2016 00:44:05 +0800
Subject: [PATCH 0625/1345] doc: usb: a3700: update porting guide for a3700
 usb otg

commit  22bdef57f302863342e185650cc3599a6321a398 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

This patch adds a section in linux porting guide about
Armada3700 USB OTG PHY driver, and explain the details
of how to modify dts files for USB OTG mode in
device-tree binding doc for it.

Change-Id: I182e2159e8410e9fe217f82c412c9a7a54310b92
Signed-off-by: jinghua <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/34126
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
Tested-by: Nadav Haklai <nadavh@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../devicetree/bindings/usb/armada3700-otg-phy.txt |   14 ++++++++++++++
 Documentation/mvebu/porting_guide.txt              |    3 +++
 2 files changed, 17 insertions(+)

diff --git a/Documentation/devicetree/bindings/usb/armada3700-otg-phy.txt b/Documentation/devicetree/bindings/usb/armada3700-otg-phy.txt
index b9c7972..03aeec0 100644
--- a/Documentation/devicetree/bindings/usb/armada3700-otg-phy.txt
+++ b/Documentation/devicetree/bindings/usb/armada3700-otg-phy.txt
@@ -2,6 +2,15 @@ USB Armada3700 OTG PHY
 
 This controller is integrated only in Armada-3700.
 
+Armada-3700 could be working in USB OTG mode which supports USB Host/Device mode
+switching dynamically. To do that, the following dts changes are required:
+- Enable both USB Host (usb3) and Device (u3d and udc) controller DT nodes.
+- In USB Host DT node, use special XHCI driver compatible string:
+  "marvell,armada-3700-xhci-otg", while the default one "generic-xhci" is for
+  USB Host only mode.
+- Enable OTG PHY TD node and make sure XHCI DT node have reference to it with
+  "usb-phy" property.
+
 Required properties:
 - compatible	: should be "marvell,armada-3700-otg"
 - reg		: Physical base address and size of the USB3 device registers map.
@@ -11,6 +20,11 @@ Optional properties:
 - vcc-supply: phandle to the regulator that provides power to the PHY.
 
 Example:
+	usb3@58000 {
+		compatible = "marvell,armada-3700-xhci-otg";
+		status = "okay";
+		usb-phy = <&otg_phy>;
+	};
 
 	otg_phy: otg@5d800 {
 		compatible = "marvell,usb-3700-otg";
diff --git a/Documentation/mvebu/porting_guide.txt b/Documentation/mvebu/porting_guide.txt
index 3c3673a..60e01c6 100644
--- a/Documentation/mvebu/porting_guide.txt
+++ b/Documentation/mvebu/porting_guide.txt
@@ -155,6 +155,9 @@ A37x0
 	- Marvell A37x0 SPI driver
 	  spi/spi-armada-3700.txt
 
+	- Marvell A37x0 USB OTG (Host/Device Switching) PHY driver
+	  Binding: usb/armada3700-otg-phy.txt
+
 A39x
 ----
 	- Marvell A39x Pinctrl driver
-- 
1.7.9.5

